### ISSUE TITLE
Segmentation fault using `producev`

### ISSUE THREAD

**Issue by rgomezcasas** [php-rdkafka/php-rdkafka#221]
* PHP version: 7.1.29
 * librdkafka version: master and v1.0.0
 * php-rdkfaka version: master and 3.1.0

Doing a basic usage, if I do
```php
$topic->produce(RD_KAFKA_PARTITION_UA, 0, 'some payload');
```

It works ok, but if instead I try to use with headers:
```php
$topic->producev(RD_KAFKA_PARTITION_UA, 0, "some payload", null, ['key' => 'value']);
```

I get a `segmentation fault`.

I read about compiling librdkafka with the `--disable-ssl --disable-sasl` options, but also I get a segfault.

Am I doing something wrong?

**Comment by Steveb-p** [comment]
@rgomezcasas could you try it against `0.11.6` build of librdfkafka?

EDIT: There is actually a test for producing messages with headers, and it should pass against `0.11.6`. I believe #220 is the culprit here (incompatibility with `1.0.0`)

https://github.com/arnaud-lb/php-rdkafka/blob/43bf4409ee6b42b78dd58b749082f679c3754c6f/tests/message_headers.phpt

**Comment by rgomezcasas** [comment]
I'll try it tomorrow! Thanks

**Comment by Steveb-p** [comment]
@rgomezcasas I've run integration tests for librdkafka `1.0.0` and lo and behold, output for `message_headers.phpt` is:
```
---- EXPECTED OUTPUT
5 messages delivered
Got message: message 0 | Headers: key: value
Got message: message 1 | Headers: key1: value1, key2: value2, key3: value3
Got message: message 2 | Headers: none
Got message: message 3 | Headers: none
Got message: message 4 | Headers: none
EOF
---- ACTUAL OUTPUT
Termsig=11
---- FAILED
```
which is segfault.

**Comment by Steveb-p** [comment]
I've pinpointed the location of segfault to come from `rdkafka_msg.c:315` from `librdkafka`. According to `git blame` it's the result of a error checking code that was introduced 04.09.2018 in commit `ec2a726415359b11294d905dfcd9de0e0039bc92` (Added fatal errors to signal non-recoverable errors to the application).

Code introduced is:
```c
if (unlikely((err = rd_kafka_fatal_error_code(rk))))
       return err;
```

Quick look into `topic.c:469` revealed that `NULL` is passed instead of `kafka_object` pointer.

```c
    err = rd_kafka_producev(
            NULL,
            RD_KAFKA_V_RKT(intern->rkt),
            RD_KAFKA_V_PARTITION(partition),
            RD_KAFKA_V_MSGFLAGS(msgflags | RD_KAFKA_MSG_F_COPY),
            RD_KAFKA_V_VALUE(payload, payload_len),
            RD_KAFKA_V_KEY(key, key_len),
            RD_KAFKA_V_HEADERS(headers),
            RD_KAFKA_V_END
    );
```

I managed to resolve your segmentation fault in my local copy of phprdkafka by acquiring `kafka_object *kafka_intern = get_kafka_object(P_ZEVAL(intern->zrk) TSRMLS_CC);` (shamelessly copied code from elsewhere in the file) before `rd_kafka_producev` is called and passing `kafka_intern->rk` instead of `NULL`.

All other failing integration tests seem to be related, since all of them result in message consumption causing timeouts.
